graphix/pattern.py

```diff
@@ -1421,13 +1421,32 @@ def run_pattern(self, backend, **kwargs):
         result = exe.run()
         return result
 
-    def perform_pauli_measurements(self, leave_input=False, use_rustworkx=False):
+    def perform_pauli_measurements(self, leave_input=False, use_rustworkx=False, ignore_pauli_with_deps=False):
         """Perform Pauli measurements in the pattern using
         efficient stabilizer simulator.
 
+        Parameters
+        ----------
+        leave_input : bool
+            Optional (`False` by default).
+            If `True`, measurements on input nodes are preserved as-is in the pattern.
+        use_rustworkx : bool
+            Optional (`False` by default).
+            If `True`, `rustworkx` is used for fast graph processing.
+            If `False`, `networkx` is used.
+        ignore_pauli_with_deps : bool
+            Optional (`False` by default).
+            If `True`, Pauli measurements with domains depending on other measures are preserved as-is in the pattern.
+            If `False`, all Pauli measurements are preprocessed. Formally, measurements are swapped so that all Pauli measurements are applied first, and domains are updated accordingly.
+
         .. seealso:: :func:`measure_pauli`
 
         """
+        if not ignore_pauli_with_deps:
+            self.standardize()
+            pauli_nodes = self.__extract_pauli_nodes()
+            first_measure_index = next((i for i, cmd in enumerate(self) if cmd.kind == CommandKind.M), len(self))
```

Comment:
Let’s use more conventional style (such as `.find`? I don’t know the precise name)

Reply:
Fixed in 42de1df. I changed the logic and now, a new sequence is built with an explicit loop that finds where the first non-Pauli measurement command is.